src/code/cutscene_camera.c

```diff
@@ -353,28 +357,576 @@ void CutsceneCamera_Reset(void) {
     sCurCsCamera->state = CS_CAM_STATE_UPDATE_ALL;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80161C20.s")
+// Linear interpolation from initial values to cmd values. Set weight to 100 to go all the way to the cmd value.
+s16 CutsceneCamera_Interp_Lerp(Vec3f* camPos, f32* camFov, s16* camRoll, CsCmdCamPoint* pointCmd, CsCmdCamMisc* miscCmd,
+                               CutsceneCameraInterp* interpState) {
+    f32 lerp;
+
+    if (interpState->type != CS_CAM_INTERP_LERP) {
+        // Initialize
+        interpState->type = CS_CAM_INTERP_LERP;
+        interpState->waypoint = 0;
+        interpState->curFrame = 0;
+        interpState->duration = 1;
+        if (camPos != NULL) {
+            interpState->initPos.x = camPos->x;
+            interpState->initPos.y = camPos->y;
+            interpState->initPos.z = camPos->z;
+        }
+
+        if (camFov != NULL) {
+            interpState->initFov = *camFov;
+        }
+
+        if (camRoll != NULL) {
+            interpState->initRoll = *camRoll;
+        }
+    }
+
+    interpState->curFrame++;
+
+    lerp = ((f32)interpState->curFrame / pointCmd->duration) * (pointCmd->weight / 100.0f);
+
+    if (camPos != NULL) {
+        VEC3F_LERPIMPDST(camPos, &interpState->initPos, &pointCmd->pos, lerp);
+    }
+
+    if (camFov != NULL) {
+        *camFov = LERPIMP(interpState->initFov, miscCmd->fov, lerp);
+    }
+
+    if (camRoll != NULL) {
+        s16 new_var;
+        s16 temp;
+
+        new_var = CAM_DEG_TO_BINANG(miscCmd->roll);
+
+        temp = (s16)(new_var - (s16)interpState->initRoll);
```

Comment:
This construction comes up three times in this file, so I feel like we could get a lot of bang for our buck by naming these two variables. If I'm understanding it right, `new_var` is essentially the "target roll", and `temp` is the difference between the target roll and the initial roll. Is that understanding right? If so, maybe `targetRoll` would work for the first one? Not sure what to call the second one though.

Reply:
Yeah new_var is targetRoll, as for temp I would think it's fake (it's a part of the lerping that's usually not named) but if not maybe rollDiffToTarget or similar yeah

Reply:
Agreed that it feels like there are some fake matches here, but I tried all variations of castings and it seems very picky about it. The temp names suggested seems fine